ujin/programmers/day06/모의고사.js

```diff
@@ -1,18 +1,17 @@
-function solution(n, lost, reserve) {
-  let answer = 0;
-  const noReserveLost = lost
-    .sort((a, b) => a - b)
-    .filter(lost => !reserve.includes(lost));
-  let hasReserve = reserve
-    .sort((a, b) => a - b)
-    .filter(reverse => !lost.includes(reverse));
+function solution(answers) {
+  const answer = [];
+  const supo1 = [1, 2, 3, 4, 5];
+  const supo2 = [2, 1, 2, 3, 2, 4, 2, 5];
+  const supo3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];
 
-  const finalLost = noReserveLost.filter(lost => {
-    const lend = hasReserve.find(reserve => Math.abs(reserve - lost) == 1);
-    if (!lend) return lost;
-    hasReserve = hasReserve.filter(reverse => reverse !== lend);
-  });
-  answer = n - finalLost.length;
+  const number1 = answers.filter((a, i) => a === supo1[i % 5]).length;
+  const number2 = answers.filter((a, i) => a === supo2[i % 8]).length;
+  const number3 = answers.filter((a, i) => a === supo3[i % 10]).length;
+  const max = Math.max(number1, number2, number3);
+
+  if (number1 === max) answer.push(1);
+  if (number2 === max) answer.push(2);
+  if (number3 === max) answer.push(3);
```

Comment:
객체 자료구조와 배열 매서드를 활용해서 하드코딩을 줄여봤습니다. 속도는 비교적 느리긴 하지만 경우의 수를 직접 입력하는 하드코딩을 줄 일 수 있습니다. 유진씨의 로직은 그대로 사용했습니다.
```suggestion
    const score = Object.values(pattern).reduce((acc, cur, index) => {
    acc[`number${index + 1}`] = answers.filter(
      (a, i) => a === cur[i % cur.length]
    ).length;
    return acc;
  }, {});

  const max = Math.max(score.number1, score.number2, score.number3);
  
  Object.values(score).forEach((num, index) => {
    if (num === max) answer.push(index + 1);
  });
```